xwiki-rendering-macros/xwiki-rendering-macro-figure/src/main/java/org/xwiki/rendering/internal/macro/figure/FigureCaptionMacro.java

```diff
@@ -86,12 +88,14 @@ public boolean supportsInlineMode()
         List<Block> result = Collections.emptyList();
 
         // If we're not inside a FigureBlock then don't do anything.
-        Block parent = context.getCurrentMacroBlock().getParent();
+        Block parent = context.getCurrentMacroBlock().getParent().getParent();
```

Comment:
Looks fragile and a bit weird, no? Also it's not the parent but the grandparent (cf variable name) ;)

Reply:
It requires explanations IMO (or better a new private method to find if we're inside a FigureBlock, i.e. traverse up till MacroMarkerBlock or something like this)

Reply:
BTW not your fault, it's my code, and it should have been better coded :) This is just revealing the fragility

Reply:
> Looks fragile and a bit weird, no? 

Completely. I wasn't even sure we should use an UnchangedContent block on this macro.
So maybe, as for MessageMacro, what we should do is to remove the UnchangedContent on this one and refactor it in another PR before trying to apply the new metadata. WDYT?